worker/provisioner/provisioner.go

```diff
@@ -132,7 +132,7 @@ var getDistributionGroupFinder = func(st *apiprovisioner.State) DistributionGrou
 }
 
 // getStartTask creates a new worker for the provisioner,
-func (p *provisioner) getStartTask(harvestMode config.HarvestMode, workerCount int) (ProvisionerTask, error) {
+func (p *provisioner) getStartTask(harvestMode config.HarvestMode, workerCount int, sshAllowList []string) (ProvisionerTask, error) {
```

Comment:
This constitutes a violation of SRP.

Specifying SSH allow lists is not the remit of the provisioner.

If the environ implements a model-wide firewaller, then we don't need specific actions during provisioning anyway - the firewaller will have modified the model-wide group already, and any new/existing instances will be affected.